staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -418,13 +438,9 @@ func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
 //
 // If you want to Poll something forever, see PollInfinite.
 func Poll(interval, timeout time.Duration, condition ConditionFunc) error {
-	return pollInternal(poller(interval, timeout), condition)
-}
-
-func pollInternal(wait WaitFunc, condition ConditionFunc) error {
-	done := make(chan struct{})
-	defer close(done)
-	return WaitFor(wait, condition, done)
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Why do you need WithCancel? Can't just just pass context.Background() ?

Reply:
Same below

Reply:
using `context.Background()` now

Reply:
using `context.Background()`